core/src/main/java/io/aiven/kafka/tieredstorage/config/CacheConfig.java

```diff
@@ -60,11 +56,11 @@ private static ConfigDef addCacheConfigs(final OptionalLong maybeDefaultSize) {
     }
 
     public CacheConfig(final Map<String, ?> props) {
-        super(addCacheConfigs(OptionalLong.empty()), props);
+        super(configDef(NO_DEFAULT_VALUE), props);
     }
 
-    public CacheConfig(final Map<String, ?> props, final long defaultSize) {
-        super(addCacheConfigs(OptionalLong.of(defaultSize)), props);
+    public CacheConfig(final Map<String, ?> props, final Long defaultSize) {
+        super(configDef(Objects.requireNonNull(defaultSize, "defaultSize cannot be null")), props);
```

Comment:
If it can't be `null`, should we keep `long` as the type here?

Reply:
Good catch. Added a Builder to conditionally pass this value when needed. PTAL.